File: shared/src/main/java/de/dreier/mytargets/shared/targets/DAIR3D.java
/*
 * MyTargets Archery
 *
 * Copyright (C) 2015 Florian Dreier
 * All rights reserved
 */
package de.dreier.mytargets.shared.targets;

import de.dreier.mytargets.shared.R;
import de.dreier.mytargets.shared.utils.Color;

import static de.dreier.mytargets.shared.targets.ZoneType.CIRCLE;
import static de.dreier.mytargets.shared.targets.ZoneType.HEART;
import static de.dreier.mytargets.shared.utils.Color.BROWN;
import static de.dreier.mytargets.shared.utils.Color.CERULEAN_BLUE;
import static de.dreier.mytargets.shared.utils.Color.GRAY;
import static de.dreier.mytargets.shared.utils.Color.GREEN;
import static de.dreier.mytargets.shared.utils.Color.LIGHT_GRAY;
import static de.dreier.mytargets.shared.utils.Color.RED;
import static de.dreier.mytargets.shared.utils.Color.TURBO_YELLOW;

public class DAIR3D extends Target3DBase {
    public static final int ID = 19;

    public DAIR3D() {
        super(ID, R.string.dair_3d);
        zones = new Zone[]{
                new Zone(CIRCLE, 26.698f, 184.448f, 250.568f, GREEN, Color.BLACK, 3),
                new Zone(CIRCLE, 62, 184.448f, 250.568f, GREEN, Color.BLACK, 3),
                new Zone(CIRCLE, 26.698f, 637.198f, 455.401f, TURBO_YELLOW, Color.BLACK, 3),
                new Zone(CIRCLE, 26.698f, 491.448f, 699.901f, TURBO_YELLOW, Color.BLACK, 3),
                new Zone(CIRCLE, 26.698f, 563.447f, 576.566f, RED, Color.BLACK, 3),
                new Zone(CIRCLE, 62, 563.447f, 576.566f, RED, Color.BLACK, 3),
                new Zone(CIRCLE, 62, 637.198f, 455.401f, TURBO_YELLOW, Color.BLACK, 3),
                new Zone(CIRCLE, 62, 491.448f, 699.901f, TURBO_YELLOW, Color.BLACK, 3),
                new Zone(CIRCLE, 208.938f, 564.375f, 577.813f, CERULEAN_BLUE, Color.BLACK, 4),
                new Zone(HEART, 500, 500, 500, LIGHT_GRAY, Color.BLACK, 3),
                new Zone(CIRCLE, 500, 500, 500, BROWN, GRAY, 5)
        };
        zonePoints = new int[][]{
                {8, 8, 12, 12, 12, 12, 11, 11, 10, 8, 0},
                {14, 14, 12, 12, 12, 12, 11, 11, 10, 8, 0},
                {8, 8, 12, 12, 12, 12, 10, 10, 10, 8, 0},
                {14, 14, 12, 12, 12, 12, 10, 10, 10, 8, 0}};
        showAsX = new boolean[]{false, false, false, false};
    }
}

File: shared/src/main/java/de/dreier/mytargets/shared/targets/models/DAIR3D.kt
/*
 * Copyright (C) 2018 Florian Dreier
 *
 * This file is part of MyTargets.
 *
 * MyTargets is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License version 2
 * as published by the Free Software Foundation.
 *
 * MyTargets is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 */
package de.dreier.mytargets.shared.targets.models

import de.dreier.mytargets.shared.R
import de.dreier.mytargets.shared.targets.Target3DBase
import de.dreier.mytargets.shared.targets.scoringstyle.ScoringStyle
import de.dreier.mytargets.shared.targets.zone.CircularZone
import de.dreier.mytargets.shared.targets.zone.HeartZone
import de.dreier.mytargets.shared.utils.Color.BLACK
import de.dreier.mytargets.shared.utils.Color.BROWN
import de.dreier.mytargets.shared.utils.Color.CERULEAN_BLUE
import de.dreier.mytargets.shared.utils.Color.GRAY
import de.dreier.mytargets.shared.utils.Color.GREEN
import de.dreier.mytargets.shared.utils.Color.LIGHT_GRAY
import de.dreier.mytargets.shared.utils.Color.RED
import de.dreier.mytargets.shared.utils.Color.TURBO_YELLOW

class DAIR3D : Target3DBase(
        id = ID,
        nameRes = R.string.dair_3d,
        zones = arrayOf(
                CircularZone(0.053396f, -0.631104f, -0.49886402f, GREEN, BLACK, 3),
                CircularZone(0.124f, -0.631104f, -0.49886402f, GREEN, BLACK, 3),
                CircularZone(0.053396f, 0.274396f, -0.089198f, TURBO_YELLOW, BLACK, 3),
                CircularZone(0.053396f, -0.017104004f, 0.399802f, TURBO_YELLOW, BLACK, 3),
                CircularZone(0.053396f, 0.12689404f, 0.15313196f, RED, BLACK, 3),
                CircularZone(0.124f, 0.12689404f, 0.15313196f, RED, BLACK, 3),
                CircularZone(0.124f, 0.274396f, -0.089198f, TURBO_YELLOW, BLACK, 3),
                CircularZone(0.124f, -0.017104004f, 0.399802f, TURBO_YELLOW, BLACK, 3),
                CircularZone(0.417876f, 0.12875f, 0.15562597f, CERULEAN_BLUE, BLACK, 4),
                HeartZone(1.0f, 0.0f, 0.0f, LIGHT_GRAY, BLACK, 3),
                CircularZone(1.0f, 0.0f, 0.0f, BROWN, GRAY, 5)
        ),
        scoringStyles = arrayOf(
                ScoringStyle(false, 8, 8, 12, 12, 12, 12, 11, 11, 10, 8, 0),
                ScoringStyle(false, 14, 14, 12, 12, 12, 12, 11, 11, 10, 8, 0),
                ScoringStyle(false, 8, 8, 12, 12, 12, 12, 10, 10, 10, 8, 0),
                ScoringStyle(false, 14, 14, 12, 12, 12, 12, 10, 10, 10, 8, 0),
                ScoringStyle(false, 14, 14, 11, 11, 10, 10, 10, 10, 10, 8, 5)
        )
) {
    companion object {
        const val ID = 19L
    }
}
